### PR TITLE
[#175919444] Add public endpoints

### PR DESCRIPTION
**Pull request by pasqualedevita**
this PR adds these parameters
1. **endpointType**: set public or private. Set to public if not necessary to deploy a container instance
1. **headers**: specify custom headers, for example secrets as azure pipeline secret variable

With custom headers we can do an authenticated request against public functions.

### CODE REVIEW

File: templates/rest-healthcheck/template.yaml [hunk 1]
```diff
@@ -34,54 +46,63 @@ parameters:
...
-        # check if vnet exists in target resource group (containerInstanceResourceGroup)
-        OUTPUT_VNET=$(az network vnet list --resource-group '${{ parameters.containerInstanceResourceGroup }}' --query "[?contains(name, '${{ parameters.containerInstanceVNet }}')].name" --output tsv)
-        if [ "$OUTPUT_VNET" = '${{ parameters.containerInstanceVNet }}' ]; then
-          echo "VNET '${{ parameters.containerInstanceVNet }}' found, continue"
-        else
-          echo "VNET '${{ parameters.containerInstanceVNet }}' not found, fail"
-          exit 1
+      inlineScript: |
+        if [ '${{ parameters.endpointType }}' = "public" ]; then
+          echo public
```
Comment (balanza) [inline templates/rest-healthcheck/template.yaml:60]: Do we need this for logging?

File: templates/rest-healthcheck/README.md [hunk 2]
```diff
@@ -14,25 +14,29 @@ resources:
...
-|containerInstanceResourceGroup|The resource group the container is mounted with||
-|containerInstanceVNet|The VNet the container is mounted onto||
-|containerInstanceSubnet|The subnet the container is mounted onto||
+| param                          | description                                                         | default                 |
+| ------------------------------ | ------------------------------------------------------------------- | ----------------------- |
+| azureSubscription              | Azure Subscription we run the script with                           |                         |
+| appName                        | The name of the app in the Azure account                            |                         |
+| endpoint                       | The endpoint to call                                                |                         |
+| endpointType                   | Set if endpoint is public or private                                |                         |
+| headers                        | Set custom headers                                                  |                         |
```
Comment (balanza) [inline templates/rest-healthcheck/README.md:37]: Please spend a some more words about the format of this value and the default behaviour

**Review by balanza — CHANGES_REQUESTED** [review]

File: templates/rest-healthcheck/README.md (hunk 2 above)
Comment (pasqualedevita) [reply templates/rest-healthcheck/README.md]: done!

File: templates/rest-healthcheck/template.yaml (hunk 1 above)
Comment (pasqualedevita) [reply templates/rest-healthcheck/template.yaml]: Yes can be useful to see it. I moved echo before if

File: templates/rest-healthcheck/README.md [hunk 3]
```diff
@@ -14,25 +14,29 @@ resources:
...
+| endpoint                       | The endpoint to call                                                         |                         |
+| endpointType                   | Set if endpoint is public or private                                         |                         |
+| headers                        | If needed add custom headers to HTTP request in KEY=VALUE format             |                         |
+| containerInstanceCpu           | Max container CPU allocation (only for private endpoint)                     | 0.5 (minimum allocable) |
+| containerInstanceMemory        | Max container memory allocation (only for private endpoint)                  | 0.5 (minimum allocable) |
+| containerInstanceResourceGroup | The resource group the container is mounted with (only for private endpoint) |                         |
+| containerInstanceVNet          | The VNet the container is mounted onto (only for private endpoint)           |                         |
+| containerInstanceSubnet        | The subnet the container is mounted onto (only for private endpoint)         |                         |
```
Comment (balanza) [inline templates/rest-healthcheck/README.md:42]: I think it's better to be explicit on what argument is required and when. Also, container-relate paramenters are always required except for vnet, aren't them?

```suggestion
| param                          | description                                                                     | default                 | required                        |
|--------------------------------|---------------------------------------------------------------------------------|-------------------------|---------------------------------|
| azureSubscription              | Azure Subscription we run the script with                                       |                         | true                            |
| appName                        | The name of the app in the Azure account                                        |                         | true                            |
| endpoint                       | The endpoint to call                                                            |                         | true                            |
| endpointType                   | Weather the endpoint is exposed on public network. Value: `public` or `private` |                         | true                            |
| headers                        | If needed add custom headers to HTTP request in KEY=VALUE format                |                         | only if `endpointType: public`  |
| containerInstanceCpu           | Max container CPU allocation                                                    | 0.5 (minimum allocable) | false                           |
| containerInstanceMemory        | Max container memory allocation                                                 | 0.5 (minimum allocable) | false                           |
| containerInstanceResourceGroup | The resource group the container is mounted with                                |                         | true                            |
| containerInstanceVNet          | The VNet the container is mounted onto                                          |                         | only if `endpointType: private` |
| containerInstanceSubnet        | The subnet the container is mounted onto                                        |                         | true                            |
```

**Review by balanza — APPROVED** [review]

File: templates/rest-healthcheck/README.md (hunk 3 above)
Comment (pasqualedevita) [reply templates/rest-healthcheck/README.md]: thank you

**Pull request merged**